### PR TITLE
4 ➡️ 5

### PR DESCRIPTION
**Pull request by chapulina**
# ➡️ Forward port

Port `ign-gazebo4` to `main`

Branch comparison: https://github.com/ignitionrobotics/ign-gazebo/compare/main...ign-gazebo4

**Note to maintainers**: Remember to **Merge** with commit (not squash-merge or rebase)

🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸

### CODE REVIEW

File: test/integration/examples_build.cc [hunk 1]
```diff
@@ -25,7 +25,7 @@
 #include "ignition/gazebo/test_config.hh"
 
 // File copied from
-// https://github.com/ignitionrobotics/ign-gui/raw/master/test/integration/ExamplesBuild_TEST.cc
+// https://github.com/ignitionrobotics/ign-gui/raw/ign-gui3/test/integration/ExamplesBuild_TEST.cc
```
Comment (adlarkin) [inline test/integration/examples_build.cc:28]: Should this link be updated? `ign-gui3` is for Citadel.

File: tutorials/distributed_simulation.md [hunk 2]
```diff
@@ -106,7 +106,7 @@ keeps all performers loaded, but performs no physics simulation.
 Stepping happens in 2 stages: the primary update and the secondaries update,
 according to the diagram below:
 
-<img src="https://raw.githubusercontent.com/ignitionrobotics/ign-gazebo/master/tutorials/files/distributed_step.png"/>
+<img src="https://raw.githubusercontent.com/ignitionrobotics/ign-gazebo/ign-gazebo3/tutorials/files/distributed_step.png"/>
```
Comment (adlarkin) [inline tutorials/distributed_simulation.md:109]:
```suggestion
<img src="https://raw.githubusercontent.com/ignitionrobotics/ign-gazebo/main/tutorials/files/distributed_step.png"/>
```

**Review by adlarkin — COMMENTED** [review]
It looks like there are still quite a bit of references to `master`:

```
$ git grep -i -n master
README.md:12:Test coverage | [![codecov](https://codecov.io/gh/ignitionrobotics/ign-gazebo/branch/master/graph/badge.svg)](https://codecov.io/gh/ignitionrobotics/ign-gazebo)
README.md:13:Ubuntu Bionic | [![Build Status](https://build.osrfoundation.org/buildStatus/icon?job=ignition_gazebo-ci-master-bionic-amd64)](https://build.osrfoundation.org/job/ignition_gazebo-ci-master-bionic-amd64)
README.md:14:Homebrew      | [![Build Status](https://build.osrfoundation.org/buildStatus/icon?job=ignition_gazebo-ci-master-homebrew-amd64)](https://build.osrfoundation.org/job/ignition_gazebo-ci-master-homebrew-amd64)
README.md:15:Windows       | [![Build Status](https://build.osrfoundation.org/buildStatus/icon?job=ignition_gazebo-ci-master-windows7-amd64)](https://build.osrfoundation.org/job/ignition_gazebo-ci-master-windows7-amd64)
src/systems/ackermann_steering/AckermannSteering.hh:100:  /// https://github.com/froohoo/ackermansteer/blob/master/ackermansteer/
src/systems/multicopter_control/LeeVelocityController.hh:49:  /// https://github.com/ethz-asl/rotors_simulator/blob/master/rotors_control/include/rotors_control/lee_position_controller.h
src/systems/multicopter_control/MulticopterVelocityControl.hh:53:  /// https://github.com/ethz-asl/rotors_simulator/blob/master/rotors_control/include/rotors_control/lee_position_controller.h
test/gtest/src/gtest-death-test.cc:250:         "https://github.com/google/googletest/blob/master/googletest/docs/"
test/gtest/src/gtest.cc:5494:// https://github.com/google/googletest/blob/master/googletest/docs/advanced.md
tutorials/erb_template.md:10:Some of them are listed below and demonstrated in this [example ERB file](https://github.com/osrf/srcsim/blob/master/worlds/unique.world.erb):
tutorials/mesh_to_fuel.md:129:![Electrical Box Test](https://raw.githubusercontent.com/ignitionrobotics/ign-gazebo/master/tutorials/files/mesh_to_fuel/model_info2.png)
tutorials/mesh_to_fuel.md:133:![Delete model](https://raw.githubusercontent.com/ignitionrobotics/ign-gazebo/master/tutorials/files/mesh_to_fuel/delete2.png)
tutorials/mesh_to_fuel.md:139:Copy [this example world code](https://github.com/ignitionrobotics/ign-gazebo/raw/master/examples/worlds/import_mesh.sdf) into a text editor and save it as `import_mesh.sdf`.
tutorials/mesh_to_fuel.md:185:![Launch sample world with mesh](https://raw.githubusercontent.com/ignitionrobotics/ign-gazebo/master/tutorials/files/mesh_to_fuel/launch_world2.png)
tutorials/point_cloud_to_mesh.md:23:![Opening the point cloud](https://raw.githubusercontent.com/ignitionrobotics/ign-gazebo/master/tutorials/files/point_cloud_to_mesh/cloudcompare2.png)
tutorials/point_cloud_to_mesh.md:37:![Min. space between points](https://raw.githubusercontent.com/ignitionrobotics/ign-gazebo/master/tutorials/files/point_cloud_to_mesh/min_space.png)
tutorials/point_cloud_to_mesh.md:43:![Point count of subsample](https://raw.githubusercontent.com/ignitionrobotics/ign-gazebo/master/tutorials/files/point_cloud_to_mesh/properties.png)
tutorials/point_cloud_to_mesh.md:51:![Selecting the new point cloud](https://raw.githubusercontent.com/ignitionrobotics/ign-gazebo/master/tutorials/files/point_cloud_to_mesh/secondcloud.png)
tutorials/point_cloud_to_mesh.md:62:![Choose Triangulation surface model](https://raw.githubusercontent.com/ignitionrobotics/ign-gazebo/master/tutorials/files/point_cloud_to_mesh/compute_normals.png)
tutorials/point_cloud_to_mesh.md:75:![Octree depth and output density selection](https://raw.githubusercontent.com/ignitionrobotics/ign-gazebo/master/tutorials/files/point_cloud_to_mesh/outputdensity.png)
tutorials/point_cloud_to_mesh.md:96:![The blob shape](https://raw.githubusercontent.com/ignitionrobotics/ign-gazebo/master/tutorials/files/point_cloud_to_mesh/blob2.png)
tutorials/point_cloud_to_mesh.md:101:![Adjusting scalar filed params](https://raw.githubusercontent.com/ignitionrobotics/ign-gazebo/master/tutorials/files/point_cloud_to_mesh/sf_display.png)
tutorials/point_cloud_to_mesh.md:107:![Display original polygons](https://raw.githubusercontent.com/ignitionrobotics/ign-gazebo/master/tutorials/files/point_cloud_to_mesh/hidden_polygons2.png)
tutorials/point_cloud_to_mesh.md:116:![Splitting the mesh](https://raw.githubusercontent.com/ignitionrobotics/ign-gazebo/master/tutorials/files/point_cloud_to_mesh/split.png)
```

There are also some references to the `ign-gazebo4` branch:
```
$ git grep -i -n gazebo4
Changelog.md:180:1. Updated source build instructions for ign-gazebo4
src/systems/ackermann_steering/AckermannSteering.hh:98:  /// https://github.com/ignitionrobotics/ign-gazebo/tree/ign-gazebo4/src/systems/diff_drive
tutorials/collada_world_exporter.md:25:Refer to the [collada_world_exporter.sdf](https://github.com/ignitionrobotics/ign-gazebo/blob/ign-gazebo4/examples/worlds/collada_world_exporter.sdf) example.
tutorials/resources.md:62:2. [GUI plugins that are installed with Ignition Gazebo](https://github.com/ignitionrobotics/ign-gazebo/tree/ign-gazebo4/src/gui/plugins)
```

File: test/integration/examples_build.cc (hunk 1 above)
Comment (chapulina) [reply test/integration/examples_build.cc:28]: This is referring to the original place where the file was, so I think it's ok to point to an older version.

**Review by adlarkin — APPROVED** [review]

**Comment by chapulina** [comment]
There are a couple of new warnings, let me address them

**Comment by adlarkin** [comment]
The warnings from homebrew CI are gone now. Should we merge this, or do we need to wait on some other checks to finish?

**Comment by adlarkin** [comment]
@chapulina, I went ahead and updated this PR with the latest changes from the `main` branch. @mjcarroll took a look, and everything seems to be okay. I'll wait to see what results we get from CI.

**Review by mjcarroll — APPROVED** [review]

**Comment by chapulina** [comment]
The `JointPositionController_TEST` failed again, and I don't see the changes from #697 on this branch, so I'll give the merge another try

**Comment by chapulina** [comment]
:crossed_fingers:

**Comment by chapulina** [comment]
Ahhhh Windows, let me fix it

**Comment by chapulina** [comment]
Pushed an attempt at fixing Windows linker errors for the GUI test

**Comment by chapulina** [comment]
Latest push disables the GUI plugin tests on Windows for now

**Comment by adlarkin** [comment]
> Latest push disables the GUI plugin tests on Windows for now

Should we create an issue so that we can re-visit this later?

**Comment by chapulina** [comment]
> Should we create an issue so that we can re-visit this later?

There are lots of disabled and failing tests on Windows, so I think at this point we don't need to track them individually yet.

**Comment by chapulina** [comment]
Ok! Windows is building. The only suspicious test failure is a timeout for `INTEGRATION_examples_build`, which I don't think I've seen before. But it looks to me that it could be due to a slow CI machine. Merging!

```
81: [ 50%] Building CXX object CMakeFiles/joy_to_twist.dir/joy_to_twist.cc.o
 81/174 Test  #81: INTEGRATION_examples_build .............................***Timeout 240.01 sec
```

**Pull request merged**